packages/flutter/test/material/theme_data_test.dart

```diff
@@ -268,6 +268,40 @@ void main() {
     expect(theme.applyElevationOverlayColor, isTrue);
   });
 
+  testWidgets('splashFactory is InkSparkle only for Android non-web when useMaterial3 is true', (WidgetTester tester) async {
+    final ThemeData theme = ThemeData.fallback().copyWith(useMaterial3: true);
```

Comment:
The problem here is that this doesn't change the default splash factory as `useMaterial3` isn't set in the constructor. To fix this:

```suggestion
    final ThemeData theme = ThemeData(useMaterial3: true);
```